### PR TITLE
Windows titlebar improvements

### PR DESCRIPTION
**Pull request by paulcbetts**
This PR fixes a few bugs around the title bar on Windows:

* Show a Restore button when the window is maximized:

image

* Allow unmaximizing by dragging the title bar

* Update the maximize button when the window is maximized via the Taskbar / System menu (Alt-space)

* Fix a bug where refreshing the window would turn it transparent

Unfortunately, in order to fix these bugs, I had to make console transparency a Mac-only feature, enabling `transparent` in Electron in Windows just results in too much Weird Stuff happening

### CODE REVIEW

File: app/index.js [hunk 1]
```diff
@@ -139,7 +139,7 @@ app.on('ready', () => installDevExtensions(isDev).then(() => {
       backgroundColor: toElectronBackgroundColor(cfg.backgroundColor || '#000'),
       // we want to go frameless on windows and linux
       frame: process.platform === 'darwin',
-      transparent: true,
+      transparent: process.platform === 'darwin',
```
Comment (juliankay) [inline app/index.js]: Will this cause an issue for people using Linux?

Comment (paulcbetts) [inline app/index.js]: transparent doesn't work on Linux unless you run with --disable-gpu and a laundry-list of other caveats :(

**Comment by paulcbetts** [comment]
How can I get this reviewed? /cc @rauchg?

**Comment by ppot** [comment]
@paulcbetts can you rebase?

**Comment by paulcbetts** [comment]
@ppot done

**Comment by paulcbetts** [comment]
@ppot Can you add me as maintainer so I can merge / review these Windows PRs? It seems as if all the maintainers are on Mac so these PRs just sit forever. I want to work on this project but like, if this is gonna happen for every PR, this is gonna go nowhere fast

**Comment by framerate** [comment]
Trying not to make ANOTHER dupe, 3 years later, but trying out Hyper again and snapping doesn't work. Am I the only person trying this one Windows in 2020? :D

**Comment by framerate** [comment]
Nevermind. Something else is funny. If you see this, try quitting and relaunching hyper. Stupid windows...

File: app/index.js (hunk 1 above)
Comment (arthurmloureiro) [reply app/index.js:142]: Hi, I accidentally enabled transparency on my hyper. This causes the windows to fail to snap into fancy areas as before. I've tried turning it off but I cannot find the index.js file anywhere. I tried also uninstalling it but it somehow finds the old configurations after I re-install it.

**Pull request merged**